hilti/runtime/include/types/enum.h

```diff
@@ -29,5 +31,38 @@ bool has_label(const T& t, const TypeInfo* ti) {
 
     return false;
 }
+
+/**
+ * Converts a signed integer value into an enum value. The value does
+ * not need to correspond to a valid label. (Internally, this is a
+ * straight-forward cast.)
+ *
+ * @param t numerical value to convert
+ * @tparam T enum type
+ */
+template<typename T>
+T from_int(int64_t n) {
+    return static_cast<T>(n);
+}
+
+/**
+ * Converts an unsigned integer value into an enum value. The value
+ * does not need to correspond to a valid label, but it cannot be
+ * larger than the maximum possible signed int64 value. (Internally,
+ * this is mostly a straight-forward cast, we just add the range
+ * check.)
+ *
+ * @param t numerical value to convert
+ * @tparam T enum type
+ * @throws InvalidValue if value exceeds range
+ */
+template<typename T>
+T from_uint(uint64_t n) {
```

Comment:
Let's add a range check of the underlying type and assert that `T` is an enum here as well.